src/isolate.rs

```diff
@@ -102,6 +102,19 @@ impl Isolate {
         }
     }
 
+    /// Schedules an exception to be thrown when returning to JavaScript.
+    ///
+    /// When an exception has been scheduled it is illegal to invoke any
+    /// JavaScript operation; the caller must return immediately and only after
+    /// the exception has been handled does it become legal to invoke JavaScript
+    /// operations.
```

Comment:
I actually thought about that as well, and I agree, it should be private. I don't know the cleanest way to do this. I could just move the function implementation to `util.rs`, but you may dislike that the `Isolate` API calls become fragmented?